### ISSUE TITLE
Rofi configuration module outdated/broken

### ISSUE THREAD

**Issue by B4rc1** [nix-community/home-manager#1593]
### Issue description

All configuration added through the rofi module does not work.
In my testing this is because it uses the old rofi configuration syntax without
``` 
configuration {
...
}
``` 
Infront (and after) and the config file does not end with a `.rasi`, which rofi then refuses to load.

#### Maintainer CC

@rycee 

#### Technical details

- system: `"x86_64-linux"`
 - host os: `Linux 5.8.7-zen1, NixOS, 20.09.1599.d105075a1fd (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.7`
 - channels(jonas): `""`
 - channels(root): `"nixos-20.09.1599.d105075a1fd"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by berbiche** [comment]
@B4rc1 would you be interested in making a PR to fix this?

**Comment by B4rc1** [comment]
unfortunately no, as I have returned to arch as my main Desktop OS and thus no longer use home-manager.

**Comment by rycee** [comment]
FWIW I use rofi under NixOS 20.09 and the configuration seems to work OK. Perhaps this is an issue that affects the version of rofi that's in NixOS unstable?

**Comment by thiagokokada** [comment]
Just did a test, took the `rofi` from unstable and did used it in the module (passing it as `programs.rofi.unstable` option).

```
$ rofi -V
Version: 1.6.1
$ rofi -show drun

(process:10269): Rofi-WARNING **: 23:34:07.765: The old Xresources based configuration format is deprecated.

(process:10269): Rofi-WARNING **: 23:34:07.765: Please upgrade: rofi -upgrade-config.
```

So yeah, I think it is still fine, but it would be better to migrate for the new format anyway since it is much nicer and I think we can use RFC42 with it.

**Comment by thiagokokada** [comment]
Since I was interested in fixing this, I did it and opened a PR: https://github.com/nix-community/home-manager/pull/1736

Didn't try to convert it to RFC42 though since it would involve a whole module rewrite, but I did some bug fixes and used the approach of manipulating attrsets instead of strings so the code should be easier to understand (of course, IMO).

This rewrite is almost 100% backwards compatible, with the exception of `programs.rofi.extraConfig` that was impossible to maintain backwards compatibility (since the config file itself changed). But even them it now uses attrsets that is much nicer to use in Nix.
